pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -452,6 +640,25 @@ public void addToError(int error) {
       _error += error;
     }
 
+    public void addToCompleted(int completed) {
+      _completed += completed;
+    }
+
+    /* Update count based on state for each task running under a HelixJob/PinotTask */
+    public void addTaskState(TaskPartitionState state) {
+      // Helix returns state as null if the task is not enqueued anywhere yet
+      if (state == null) {
+        // task is not yet assigned to a participant
+        addToWaiting(1);
+      } else if (state.equals(TaskPartitionState.INIT) || state.equals(TaskPartitionState.RUNNING)) {
+        addToRunning(1);
+      } else if (state.equals(TaskPartitionState.TASK_ERROR)) {
+        addToError(1);
+      } else if (state.equals(TaskPartitionState.COMPLETED)) {
+        addToCompleted(1);
+      }
```

Comment:
Let's add a catch all state (e.g. `Unknown`) in case we got other states

Reply:
added